mmocr/visualization/kie_visualizer.py

```diff
@@ -0,0 +1,505 @@
+        val_index = np.array(list(set(pairs[1])))
+        key_texts = [texts[i] for i in key_index]
+        val_texts = [texts[i] for i in val_index]
+
+        self.set_image(image)
+        if key_texts:
+            self.draw_texts(
+                key_texts, (bboxes[key_index, :2] + bboxes[key_index, 2:]) / 2,
+                colors='k',
+                horizontal_alignments='center',
+                vertical_alignments='center')
+        if val_texts:
+            self.draw_texts(
+                val_texts, (bboxes[val_index, :2] + bboxes[val_index, 2:]) / 2,
+                colors='k',
+                horizontal_alignments='center',
+                vertical_alignments='center')
+        self.draw_arrows(
+            x_datas,
+            y_datas,
+            colors=arrow_colors,
+            line_widths=0.3,
+            arrow_tail_widths=0.05,
+            arrow_head_widths=5,
+            overhangs=1,
+            arrow_shapes='full')
+        return self.get_image()
+
+    def _draw_instances(
+        self,
+        image: np.ndarray,
+        bbox_labels: Union[np.ndarray, torch.Tensor],
+        bboxes: Union[np.ndarray, torch.Tensor],
+        polygons: Sequence[np.ndarray],
+        edge_labels: Union[np.ndarray, torch.Tensor],
+        texts: Sequence[str],
+        class_names: Dict,
+        is_openset: bool = False,
+        arrow_colors: str = 'g',
+    ) -> np.ndarray:
+        """Draw instances on image.
+
+        Args:
+            image (np.ndarray): The origin image to draw. The format
+                should be RGB.
+            bbox_labels (np.ndarray, torch.Tensor): The bbox labels to draw.
+                The shape of bbox_labels should be (N,), where N is the
+                number of texts.
+            bboxes (np.ndarray, torch.Tensor): The bboxes to draw. The shape of
+                bboxes should be (N, 4), where N is the number of texts.
+            polygons (Sequence[np.ndarray]): The polygons to draw. The length
+                of polygons should be the same as the number of bboxes.
+            edge_labels (np.ndarray, torch.Tensor): The edge labels to draw.
+                The shape of edge_labels should be (N, N), where N is the
+                number of texts.
+            texts (Sequence[str]): The texts to draw. The length of texts
+                should be the same as the number of bboxes.
+            class_names (dict): The class names for bbox labels.
+            is_openset (bool): Whether the dataset is openset. Default: False.
+        """
+        img_shape = image.shape[:2]
+        empty_shape = (img_shape[0], img_shape[1], 3)
+
+        if polygons:
+            polygons = [polygon.reshape(-1, 2) for polygon in polygons]
+        if polygons:
+            image = self._draw_polygons(
+                self, image, polygons, filling=True, colors=PALETTE)
+        else:
+            image = self._draw_bboxes(
+                self, image, bboxes, filling=True, colors=PALETTE)
```

Comment:
- The fact that drawing polygons has a higher priority than drawing bboxes hasn't been documented in docstring.
- Also, users won't be able to know one of `polygon` and `bbox` can be None from docstring and type hints.

So my suggestion are:
- Set the default values of `polygons` and `bboxes` to be None
- update docstr & type hints
- Raise an error if both inputs are None; raise a warning if both contain something
- `if polygon` -> `if polygon is not None`